cmake/macros/CompositeSources.cmake

```diff
@@ -47,35 +44,69 @@ function(composite_sources target sources_var)
     return()
   endif()
 
+  # Get rid of the periods in the extensions and create each list of composite
+  # sources.
+  set(extensions "")
+  foreach(extension ${COMPOSITE_SOURCE_EXTENSIONS})
+    string(MAKE_C_IDENTIFIER extension ${extension})
```

Comment:
Is `MAKE_C_IDENTIFIER` necessary/appropriate here? For the extensions that are eligible for compositing, we want to keep the extension exactly the same and not make any perturbations. (The `MAKE_C_IDENTIFIER` operation is generally used for naming C symbols; it's an odd choice for filename sanitation.)

Reply:
I'm using it here to create lists that will be used as buckets for each file extension, so I'll only have to loop over the source list once instead of however many source types there are. Then they're combined into a single, now sorted list.